### DISCUSSION TITLE
RPN output and flavor?

### DISCUSSION THREAD

**Discussion by RobertZenz** [ezylang/EvalEx#282] [General]
I've been looking at the RPN output that it is being generated (because of another project), and two questions popped into my mind:

"(" is used to denote the beginning of an argument list, however, "plain" RPN does not seem to state that. Is that from a certain RPN "flavor" or just for EvalEx?

Am I seeing this correct that the RPN output with multiple arguments to a function, the arguments are only separated by two arguments following each other? Assuming the following:
    
    MAX(1+2+5, 3*4, 5/8)
    ( 1 2 + 5 + 3 4 * 5 8 / MAX

As far as I can see, the only way to separate the arguments is because two numbers are following each other "in the middle of the expression".

**Comment by uklimaschewski** [comment]
Hi @RobertZenz ,

The open open parenthesis is used to mark the start (or end in evaluation stack) of the parameter list, this trick was introduced to support functions with a variable numbers of parameters.
So yes, this is something special in EvalEx.

Each parameter token is assumed to be an expression on the stack, when it comes to evaluating the expression.
If you read it from right to left:

- Function Max
- Parameter 3: Division of 5 and 8
- Parameter 2: Multiplication of 3 and 4
- Parameter 1: Addition of 5 and the addition of 1 and 2
- End Parameter

**Reply by RobertZenz** [reply]
I see, thank you!
